### ISSUE TITLE
Image Editing: Ability to rotate images

### ISSUE THREAD

**Issue by krissy** [empowerhack/DrawMyLife-Service#88]
# What

Allow the user to rotate images on upload/edit and add interaction buttons
# How
- Implement rotation functionality with our existing file management gem [Paperclip](https://github.com/thoughtbot/paperclip#custom-attachment-processors) - We can add some rough link/s to trigger the rotation on the frontend so you're not blocked by UI decisions
- Design and implement an Image Editing UI (inline buttons vs modal?)
# Considerations

Liase with team on Slack re UI

**Comment by rocioleon** [comment]
The solution is based on the research about Ephox TextBoox.io and Wix.com edit image features. There are attached a wireframe to be reviewed. 

![ephox_research_](https://cloud.githubusercontent.com/assets/18365496/18893652/b404b5b4-8506-11e6-83f2-5ba3f8c13351.JPG)

https://www.ephox.com/blog/from-the-ux-desk-road-testing-inline-image-editing/

**Comment by rocioleon** [comment]
Here is a link to see the TextBox.io documentation in case this is helpful 
http://docs.ephox.com/display/tbio/Getting+Started

**Comment by rocioleon** [comment]
After feedback, here is the amended wireframe for desktop. The idea is after having selected the file, the user can click on the thumbnail to open the edit popup module to rotate, and crop(rotate or the image. Then apply changes or cancel to close the popup. 
![in-line_image_editing-1024px 5](https://cloud.githubusercontent.com/assets/18365496/19054460/6896a1f4-89b7-11e6-9461-3129fd44f1a7.png)
![in-line_image_editing-1024px_popup 6](https://cloud.githubusercontent.com/assets/18365496/19054461/689beba0-89b7-11e6-9bef-48bba81c48a8.png)

**Comment by rocioleon** [comment]
This is the version for the phone device
![in-line_image_editing-320px 7](https://cloud.githubusercontent.com/assets/18365496/19054567/db5e868e-89b7-11e6-98d8-0b7a51ea1e5a.png)

**Comment by rocioleon** [comment]
Any feedback or question...

**Comment by marizoldi** [comment]
What if we placed the crop/rotate/ok functions in a small rounded menu that is an overlay onthe image (currently I put it on the right side). The cancel function can be just clicking outside the overlay box. I have made an example below.

I have used Material icons which are free and super helpful for dev too: https://design.google.com/icons

They inlcude this rotate and crop function which I think is useful to add to the menu ..

![crop-rotate-01](https://cloud.githubusercontent.com/assets/16060989/19155582/ea66e38a-8bd6-11e6-8c8e-72f5d3cea52a.png)
![crop-rotate-02](https://cloud.githubusercontent.com/assets/16060989/19155591/eee77a0a-8bd6-11e6-938e-19c343316ca8.png)

**Comment by rocioleon** [comment]
Thanks! Let's talk about it.

**Comment by CathMollie** [comment]
Both of these ideas are good. We could A/B test them maybe - with our existing piloters in Serbia or with our EH community?
@rocioleon what do you think?

**Comment by tanyapowell** [comment]
@rocioleon for the mobile version, would a user still have to double click on an image to get the image editing options to appear in the overlay? 
Also, is it worth having a button so there is a clear CTA instead of having a label saying `(Double click on image to edit)`? So that it's _really really_ obvious that option is available?

**Comment by krissy** [comment]
@CathMollie @rocioleon @EChesters @tanyapowell @marizoldi Who owns the UX decisions for this ticket and #89? Now we're moving towards HDX dataset integration we need to prioritise the quality of our data, so ability to rotate (then crop secondary) could now become a priority, at least for Super Admins to start for post-editing. Should we resume convo?

**Comment by CathMollie** [comment]
@krissy From memory, @EChesters was looking at crop (#88) and @rocioleon was looking at rotate (#89). We parked both. If picking up again, agree we need to get wireframes to a point we're happy with. Both Elizabeth and Rocio are working on other stuff (Upload form and Branding) but a lot of work has been done, so perhaps we can either take decisions in a team call, OR buddy up front-end/UX to work together (with Elizabeth & Rocio consulting but front-end leading).

**Comment by CathMollie** [comment]
Cath comments on Slack, 16 March:
- Consolidate #88 and #89 into one ticket (this one)
- UX Team to approve @marizoldi's suggestion of using Material icons (which are OS)
- Separate exploratory dev work into its own ticket
@rocioleon to review and comment by 19 March

**Comment by rocioleon** [comment]
Working on this week as wireframe needs to be amend + Google Material design suggested by @marizoldi + Upload New Drawing Form design by @EChesters 
Then,
-#89 will be updated accordingly @EChesters
